Source/components/SymbolInstance.js

```diff
@@ -0,0 +1,115 @@
+import { DefinedPropertiesKey } from '../WrappedObject'
+import { Layer } from './Layer'
+import { Rectangle } from '../Rectangle'
+import { Types } from '../enums'
+import { Factory } from '../Factory'
+import { wrapObject } from '../wrapNativeObject'
+import { toArray } from '../utils'
+import { Override } from './Override'
+
+/**
+ * A Sketch symbol instance.
+ */
+export class SymbolInstance extends Layer {
+  /**
+   * Make a new symbol instance.
+   */
+  constructor(master = {}) {
+    if (!master.sketchObject) {
+      // eslint-disable-next-line no-param-reassign
+      master.sketchObject = Factory.createNative(SymbolInstance)
+        .alloc()
+        .initWithFrame(new Rectangle(0, 0, 100, 100).asCGRect())
+    }
+    super(master)
+  }
+
+  // Replaces the instance with a group that contains a copy of the Symbol this instance refers to. Returns null if the master contains no layers instead of inserting an empty group
+  detach() {
+    const group = this._object.detachByReplacingWithGroup()
+
+    if (group) {
+      return wrapObject(group)
+    }
+
+    return null
+  }
+
+  setOverrideValue(override, value) {
+    this._object.setValue_forOverridePoint(
+      value,
+      wrapObject(override).sketchObject.overridePoint()
+    )
+    return this
+  }
+}
+
+SymbolInstance.type = Types.SymbolInstance
+SymbolInstance[DefinedPropertiesKey] = { ...Layer[DefinedPropertiesKey] }
+Factory.registerClass(SymbolInstance, MSSymbolInstance)
+
+SymbolInstance.define('symbolId', {
+  depends: 'parent',
+  get() {
+    return String(this._object.symbolID())
+  },
+  set(id) {
+    // we need to find the symbol master and change the master,
+    // it's not enough to just call `this._object.setSymbolID`
+    const parentPage = this._object.parentPage()
+    if (!parentPage) {
+      throw new Error(
+        'A symbol instance needs to be inserted in a page before setting the symbolId'
+      )
+    }
+    const master = parentPage.documentData().symbolWithID(id)
```

Comment:
Should we protect this? What would happen if we were given an invalid `id` and `master` came back `nil`? Would the instance be left in a bad state or could we crash trying to wrap / unwrap it?

Reply:
Yeah that would probably be bad. There are a lot of places where we should probably add more protection and throw JS errors instead of messing the file.

Mind if I do it all in another PR?

Reply:
> Mind if I do it all in another PR?

Go for it